### ISSUE TITLE
Postgres broken?

### ISSUE THREAD

**Issue by endophage** [golang-migrate/migrate#93]
I may be missing something but I'm trying to run a single migration against an empty postgres database and I get the following behaviour:

```
$> migrate -path postgres -database postgresql://the_user@hostname/db_name -verbose up 1
2018/08/28 21:57:06 Start buffering 20180827162119/u initial
2018/08/28 21:57:06 Read and execute 20180827162119/u initial
2018/08/28 21:57:06 error: pq: relation "schema_migrations" does not exist in line 0: TRUNCATE "schema_migrations"
$> migrate -path postgres -database postgresql://the_user@hostname/db_name -verbose up 1
2018/08/28 22:00:19 error: Dirty database version 20180827162119. Fix and force version.
```

In the first instance I would have expected migrate to create the `schema_migrations` table and I see code to do that. I saw another closed issue with the same error and it was suggested the `search_path` should be set. I tried this and still get the exact same behaviour as above.

Also worth noting, after running the command the first time the database appears to be empty. Only the public schema exists and it has no tables in it. After running the same command again, most but not all of the tables are created and as you can see, there's no useful error on which specific query in the migration failed.

**Comment by endophage** [comment]
Version 3.4.0 installing into the postgres:9.6 image from docker hub. Database was already created.

I finally got it running, seems to be some kind of issue with roles? I don't have any explicit changes of role in the migrations but the ownership of various tables (not `schema_migrations`) was changed as part of the migration. Postgres is not providing particularly helpful errors but changing the role (in the connection string) I ran the migrations as to the same role that ultimately owned most of the tables (and as such, `schema_migrations` was owned by that same role), seemed to solve the issue.
